src/modelmanager.cpp

```diff
@@ -869,6 +869,35 @@ Status ModelManager::tryReloadGatedModelConfigs(std::vector<ModelConfig>& gatedM
     return firstErrorStatus;
 }
 
+#ifdef _WIN32
+class FileHandle {
+public:
+    FileHandle(const std::string& filename = "") : filename_(filename) {
```

Comment:
Why empty filename is allowed?

Reply:
This could potentially crash as well in that case as temporary string is created.